src/overlays/actors/ovl_En_Ru1/z_en_ru1.c

```diff
@@ -19,417 +135,2276 @@ const ActorInit En_Ru1_InitVars = {
+    this->action = 24;
+    this->drawConfig = 1;
+    this->actor.flags |= 9;
+}
+
+void func_80AEF2D0(EnRu1* this, GlobalContext* globalCtx) {
+    s32 cond;
+
+    func_80AEEF68(this, globalCtx);
+    EnRu1_FrameUpdateMatrix(this);
+    func_80AEAE1C(this);
+    func_80AEAC10(this, globalCtx);
+    func_80AEAECC(this, globalCtx);
+    cond = func_80AEE264(this, globalCtx);
+    func_80AED624(this, globalCtx);
+    func_80AEF170(this, globalCtx, cond);
+}
+
+void func_80AEF354(EnRu1* this, GlobalContext* globalCtx) {
+    func_80AEEFEC(this, globalCtx);
+    EnRu1_FrameUpdateMatrix(this);
+    func_80AEAE1C(this);
+    func_80AEAECC(this, globalCtx);
+    func_80AEF188(this, globalCtx);
+}
+
+void func_80AEF3A8(EnRu1* this, GlobalContext* globalCtx) {
+    s32 something;
+
+    func_80AED83C(this);
+    something = EnRu1_FrameUpdateMatrix(this);
+    func_80AEF080(this);
+    func_80AEAE1C(this);
+    func_80AEAECC(this, globalCtx);
+    func_80AEF1F0(this, globalCtx, something);
+}
+
+void func_80AEF40C(EnRu1* this) {
+    SkelAnime* skelAnime = &this->skelAnime;
+
+    if ((func_800A56C8(skelAnime, 2.0f)) || (func_800A56C8(skelAnime, 7.0f)) || (func_800A56C8(skelAnime, 12.0f)) ||
+        (func_800A56C8(skelAnime, 18.0f)) || (func_800A56C8(skelAnime, 25.0f)) || (func_800A56C8(skelAnime, 33.0f))) {
+        func_80078914(&this->actor.unk_E4, 0x803);
+    }
+}
+
+void func_80AEF4A8(EnRu1* this, GlobalContext* globalCtx) {
+    Audio_PlaySoundAtPosition(globalCtx, &this->actor.unk_E4, 0x14, NA_SE_VO_RT_FALL);
+}
+
+void func_80AEF4E0(EnRu1* this) {
+    if (func_800A56C8(&this->skelAnime, 5.0f)) {
+        func_80078914(&this->actor.unk_E4, NA_SE_VO_RT_LAUGH_0);
+    }
+}
+
+void func_80AEF51C(EnRu1* this) {
+    func_80078914(&this->actor.unk_E4, NA_SE_VO_RT_THROW);
+}
+
+void func_80AEF540(EnRu1* this) {
+    if (func_80AEB104(this) == 2) {
+        func_80AEAEA4(this, 3);
+        func_80AEAEB8(this, 2);
+        if (this->skelAnime.mode != 2) {
+            func_80AEB264(this, &D_06004BF0, 2, -8.0f, 0);
+            func_80AEF51C(this);
+        }
+    }
+}
+
+void func_80AEF5B8(EnRu1* this) {
+    f32 animCurrentFrame;
+    if (D_80AF1938 == 0) {
+        animCurrentFrame = this->skelAnime.animCurrentFrame;
+        if (animCurrentFrame >= 60.0f) {
+            func_80AEAEA4(this, 3);
+            func_80AEAEB8(this, 0);
+            func_80AED57C(this);
+            D_80AF1938 = 1;
+        }
+    }
+}
+
+void func_80AEF624(EnRu1* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->actor;
+    f32 frameCount;
+    CsCmdActorAction* csCmdActorAction;
+    CsCmdActorAction* csCmdActorAction2;
+    s16 newRotTmp;
+
+    if (func_80AEAFE0(globalCtx, 1, 3)) {
+        frameCount = SkelAnime_GetFrameCount(&D_060063F4.genericHeader);
+        // this weird part with the redundant variable is necessary to match for some reason
+        csCmdActorAction2 = globalCtx->csCtx.actorActions[3];
+        csCmdActorAction = csCmdActorAction2;
```

Comment:
shot in the dark, but does a struct copy work here? i dont imagine it would have the weird temp stuff but who knows

Reply:
You mean for this?
```
thisx->posRot.pos.x = csCmdActorAction->startPos.x;
thisx->posRot.pos.y = csCmdActorAction->startPos.y;
thisx->posRot.pos.z = csCmdActorAction->startPos.z;
```
these are unfortunately different types. I think I tried all sorts of struct copying and type differences and stuff...this one took a while to match